packages/components/infinite-scroll-area/tests/infinite-scroll-area.test.tsx

```diff
@@ -41,4 +33,76 @@ describe("<InfiniteScrollArea />", () => {
     )
     await a11y(container)
   })
+
+  test("InfiniteScrollArea renders with initialLoad correctly", async () => {
+    const MyComponent = () => {
+      const [count, setCount] = useState<number>(50)
+
+      return (
+        <InfiniteScrollArea
+          onLoad={async ({ index, finish }) => {
+            setCount((prev) => prev + 50)
+
+            if (index >= 5) finish()
+          }}
+          initialLoad
+          loading={<>Loading…</>}
+        >
+          {Array(count)
+            .fill(0)
+            .map((_, index) => (
+              <div key={index}>{index}</div>
+            ))}
+        </InfiniteScrollArea>
+      )
+    }
+
+    const { getByText } = render(<MyComponent />)
+    expect(getByText("51")).toBeInTheDocument()
+  })
+
+  test("InfiniteScrollArea renders with Reverse correctly", async () => {
+    const MyComponent = ({
+      isReverse,
+    }: Pick<InfiniteScrollAreaProps, "isReverse">) => {
+      const rootRef = useRef<HTMLDivElement>(null)
+
+      return (
+        <div ref={rootRef}>
+          <InfiniteScrollArea
+            isReverse={isReverse}
+            rootRef={rootRef}
+            loading={<>Loading…</>}
+          >
+            {Array(50)
+              .fill(0)
+              .map((_, index) => (
+                <div key={index}>{index}</div>
+              ))}
+          </InfiniteScrollArea>
+        </div>
+      )
+    }
+
+    const { container, rerender } = render(<MyComponent isReverse={true} />)
+    fireEvent.scroll(container, {
+      target: {
+        scrollTop: 1000,
+      },
+    })
+    await waitFor(async () => {
+      expect(container.scrollTop).toBe(1000)
+    })
```

Comment:
Unnecessary `async`
```ts
    await waitFor(() => expect(container.scrollTop).toBe(1000))
```

Reply:
but `await` is wait to get promise return. so if remove async, no need await.

Reply:
This is not asynchronous, so `async` is meaningless.
```ts
expect(container.scrollTop).toBe(1000)
```
Assertions within `waitFor` will be executed multiple times until they succeed.